### PR TITLE
Alerts | Route-Specific Alerts on Real-Time Tracking Page

### PR DESCRIPTION
**Pull request by i0sea**
#### Summary of changes

**Asana Ticket:** [Alerts | Route-Specific Alerts on Real-Time Tracking Page](https://app.asana.com/0/385363666817452/1204570522422001/f)

---
Render route specific alerts above departures schedule

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

### CODE REVIEW

File: apps/site/assets/ts/stop/components/DepartureList.tsx [hunk 1]
```diff
@@ -28,10 +37,28 @@ const DepartureList = ({
...
+    alertsAffectingBothDirectionsArray,
+    alertsDirectionArray
+  ).filter(alert => {
+    return ["detour", "suspension", "shuttle"].includes(alert.effect);
+  });
   // TODO: handle no predictions or schedules case and predictions only case
   return (
     <>
+      <Alerts alerts={allAlerts} />
       {schedules.length && (
```
Comment (kotva006) [inline apps/site/assets/ts/stop/components/DepartureList.tsx:62]: question:  Do we want to display schedules if the route is suspended?  Will we have schedules to display then?

**Review by kotva006 — COMMENTED** [review]
Just some questions

File: apps/site/assets/ts/stop/components/DepartureList.tsx (hunk 1 above)
Comment (i0sea) [reply apps/site/assets/ts/stop/components/DepartureList.tsx:62]: If the route is suspended it's probably best not show schedules. Will add a check. They show on `dev-green` now and it's a bit confusing to see a closed stop with schedules https://dev-green.mbtace.com/stops/15566

File: apps/site/assets/ts/stop/components/DepartureList.tsx [hunk 2]
```diff
@@ -28,40 +38,61 @@ const DepartureList = ({
   );
 
   let departures: DepartureInfo[] = [];
+  const groupedAlerts = alertsByRoute(alerts);
+  const alertsForRoute = groupedAlerts[route.id] || [];
 
+  const alertsByDirectionObj = alertsByDirectionId(alertsForRoute);
+  const alertsAffectingBothDirectionsArray = alertsAffectingBothDirections(
+    alertsForRoute
+  );
```
Comment (KaylaBrady) [inline apps/site/assets/ts/stop/components/DepartureList.tsx:47]: thought (non-blocking): It seems like it might be tricky to remember to do both of these calls wherever we need to get all alerts by direction id. Maybe adding a helper like `allAlertsForDirection(directionId)` would simplify rather than needing to call both `alertsByDirectionId` and `alertsAffectingBothDirections`

File: apps/site/assets/ts/stop/components/DepartureList.tsx [hunk 3]
```diff
@@ -28,40 +38,61 @@ const DepartureList = ({
...
+
+  const alertsDirectionArray = alertsByDirectionObj[directionId]
+    ? alertsByDirectionObj[directionId]
+    : [];
+
+  const allAlerts = concat(
+    alertsAffectingBothDirectionsArray,
+    alertsDirectionArray
+  ).filter(alert => {
+    return ["detour", "suspension", "shuttle"].includes(alert.effect);
```
Comment (KaylaBrady) [inline apps/site/assets/ts/stop/components/DepartureList.tsx:57]: suggestion (non-blocking) : It looks like detour, suspension, and shuttle are considered [here](https://github.com/mbta/dotcom/blob/master/apps/site/assets/ts/stop/components/DepartureTimes.tsx#L38) and in https://github.com/mbta/dotcom/pull/1609 as high-priority alerts. Maybe it would be helpful for consistency to define a function along the lines of `isHighPriorityAlert` that could be used in both of these places so that the logic for what alerts should be displayed is shared.

File: apps/site/assets/ts/stop/components/DepartureList.tsx [hunk 4]
```diff
@@ -28,40 +38,61 @@ const DepartureList = ({
...
   // TODO: handle no predictions or schedules case and predictions only case
   return (
     <>
-      {schedules.length && (
-        <div className="stop-departures departure-list-header">
-          <div className={`departure-card__route ${routeBgClass(route)}`}>
-            <div>
-              {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
-              {routeName(route)}
+      <Alerts alerts={allAlerts} />
```
Comment (KaylaBrady) [inline apps/site/assets/ts/stop/components/DepartureList.tsx:62]: question: Can there be more than one alert here? Does the order matter?

File: apps/site/assets/ts/stop/components/DepartureList.tsx [hunk 5]
```diff
@@ -28,40 +38,61 @@ const DepartureList = ({
...
+  const alertsDirectionArray = alertsByDirectionObj[directionId]
+    ? alertsByDirectionObj[directionId]
+    : [];
+
+  const allAlerts = concat(
+    alertsAffectingBothDirectionsArray,
+    alertsDirectionArray
+  ).filter(alert => {
+    return ["detour", "suspension", "shuttle"].includes(alert.effect);
+  });
```
Comment (KaylaBrady) [inline apps/site/assets/ts/stop/components/DepartureList.tsx:58]: question: The acceptance criteria of this ticket distinguishes between some differences in alerts that should be displayed, for example: 

> * If there is a shuttle alert associated with the stop that is currently active or upcoming, it should be shown on the realtime tracking page.
> * If there is a shuttle alert elsewhere along the route that is currently active, it should be shown on the realtime tracking page.

Does this list of alerts account for this active or upcoming distinction for route-specific alerts?
Will the stop-level alerts be addressed as a separate PR?

**Review by KaylaBrady — COMMENTED** [review]
A few questions!

File: apps/site/assets/ts/stop/components/DepartureList.tsx (hunk 4 above)
Comment (i0sea) [reply apps/site/assets/ts/stop/components/DepartureList.tsx:62]: There can be more than one alert based on the `concat` ordering the route level alerts will appear first than stop level.

File: apps/site/assets/ts/stop/components/DepartureList.tsx (hunk 5 above)
Comment (i0sea) [reply apps/site/assets/ts/stop/components/DepartureList.tsx:58]: I believe the stop level alerts are already captured in `allAlerts`. If a stop is  closed wouldn't that also show up on the route as a detour or shuttle? Oh there is no distinction atm.
